e2e/actions/actions.go

```diff
@@ -34,214 +106,248 @@ func (c *actionTests) actionRun(t *testing.T) {
+			argv: []string{env.ImagePath, "test", "-f", "/scif/apps/bar/filebar.exec"},
 			exit: 0,
 		},
 		{
 			name: "ScifTestfolderOrg",
-			argv: []string{c.env.ImagePath, "test", "-d", "/scif/data/foo/output"},
+			argv: []string{env.ImagePath, "test", "-d", "/scif/data/foo/output"},
 			exit: 0,
 		},
 		{
 			name: "ScifTestfolderOrg",
-			argv: []string{c.env.ImagePath, "test", "-d", "/scif/data/foo/input"},
+			argv: []string{env.ImagePath, "test", "-d", "/scif/data/foo/input"},
 			exit: 0,
 		},
 		{
-			name: "WorkdirContain",
-			argv: []string{"--contain", c.env.ImagePath, "test", "-f", tmpfile.Name()},
+			name: "ContainOnly",
+			argv: []string{"--contain", env.ImagePath, "test", "-f", tmpfilePath},
 			exit: 1,
 		},
 		{
-			name: "Workdir",
-			argv: []string{"--workdir", "testdata", c.env.ImagePath, "test", "-f", tmpfile.Name()},
+			name: "WorkdirOnly",
+			argv: []string{"--workdir", testdata, env.ImagePath, "test", "-f", tmpfilePath},
+			exit: 1,
+		},
+		{
+			name: "WorkdirContain",
+			argv: []string{"--workdir", testdata, "--contain", env.ImagePath, "test", "-f", tmpfilePath},
 			exit: 0,
 		},
 		{
 			name: "PwdGood",
-			argv: []string{"--pwd", "/etc", c.env.ImagePath, "true"},
+			argv: []string{"--pwd", "/etc", env.ImagePath, "true"},
 			exit: 0,
 		},
 		{
 			name: "Home",
-			argv: []string{"--home", pwd + "testdata", c.env.ImagePath, "test", "-f", tmpfile.Name()},
+			argv: []string{"--home", testdata, env.ImagePath, "test", "-f", tmpfile.Name()},
 			exit: 0,
 		},
 		{
 			name: "HomePath",
-			argv: []string{"--home", "/tmp:/home", c.env.ImagePath, "test", "-f", "/home/" + testfile.Name()},
+			argv: []string{"--home", testdataTmp + ":/home", env.ImagePath, "test", "-f", homePath},
 			exit: 0,
 		},
 		{
 			name: "HomeTmp",
-			argv: []string{"--home", "/tmp", c.env.ImagePath, "true"},
+			argv: []string{"--home", "/tmp", env.ImagePath, "true"},
 			exit: 0,
 		},
 		{
 			name: "HomeTmpExplicit",
-			argv: []string{"--home", "/tmp:/home", c.env.ImagePath, "true"},
+			argv: []string{"--home", "/tmp:/home", env.ImagePath, "true"},
+			exit: 0,
+		},
+		{
+			name: "UserBindVarTmp",
+			argv: []string{"--bind", testdataTmp + ":/var/tmp", env.ImagePath, "test", "-f", vartmpfilePath},
 			exit: 0,
 		},
 		{
-			name: "UserBind",
-			argv: []string{"--bind", "/tmp:/var/tmp", c.env.ImagePath, "test", "-f", "/var/tmp/" + testfile.Name()},
+			name: "UserBindTmp",
+			argv: []string{"--bind", testdataTmp + ":/tmp", env.ImagePath, "test", "-f", tmpfilePath},
 			exit: 0,
 		},
 		{
+			// as /root directory is always present in images, this test
+			// could not run with RootProfile, RootUserNamespaceProfile and
+			// FakerootProfile
 			name: "NoHome",
-			argv: []string{"--no-home", c.env.ImagePath, "ls", "-ld", user.Dir},
+			argv: []string{"--no-home", env.ImagePath, "ls", "-ld", u.Dir},
 			exit: 2,
+			excludedProfiles: []e2e.SingularityProfile{
```

Comment:
Yeah, this is clearly not a good example, was lazy so maybe it's better to take the approach of grouping similar tests by eligible "profiles" instead of adding a filter field in test table ?